.ado/scripts/xcode_select_current_version.sh

```diff
@@ -7,4 +7,4 @@ else
 fi
 
 echo "Running command: sudo xcode-select --switch $XCODE_PATH"
-sudo xcode-select --switch "$XCODE_PATH"
\ No newline at end of file
+/usr/bin/sudo /usr/bin/xcode-select --switch "$XCODE_PATH"
```

Comment:
Is this just a preference to be exact here? No problems with it, but was curious if it caused issues or not

Reply:
This is proactive.  I'm not aware of issues with `sudo` and `xcode-select` specifically, but `python` invoked without an absolute path certainly has caused us some problems.  Thus a general rule: we always want to be as specific as possible when running system tools, for predictable results that don't depend on the user's PATH or whatever else they might have installed.  The only exceptions I have made are where there's no official blessed installation of a particular tool, i.e. `gem`, `yarn` and `pod`.

Reply:
You can see the "official" installed software for ADO macOS VMs here: https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md

Reply:
Though I'm not seeing paths listed either in there

Reply:
I'm calling this script directly from the nuget pack script to avoid having to double-specify the officially supported Xcode path.  So the audience is no longer limited to ADO machines.

Reply:
Oh right, my bad